kratos/containers/pointer_vector_set.h

```diff
@@ -178,29 +178,18 @@ class PointerVectorSet final
      */
     TDataType& operator[](const key_type& Key)
     {
-        ptr_iterator sorted_part_end;
-
-        if (mData.size() - mSortedPartSize >= mMaxBufferSize) {
-            Sort();
-            sorted_part_end = mData.end();
+        auto itr_pos = std::lower_bound(mData.begin(), mData.end(), Key, CompareKey());
+        if (itr_pos == mData.end()) {
+            // insert a new element with id.
+            mData.push_back(TPointerType(new TDataType(Key)));
+            return back();
+        } else if (EqualKeyTo(Key)(*itr_pos)) {
```

Comment:
isn t this second check always verified?

Reply:
`std::lower_bound` returns an iterator for the first element which is greater than or equal to the value being checked. So, if the returned `iterator` is equal to the given key, then it returns the exiting element, otherwise a new element is created with the passed key and the `iterator` of that is returned.

If possible, I would like to remove the element creation in this method, but this behaviour is used heavily, hence I kept the behavious as it is.